pyacq/core/sharedarray.py

```diff
@@ -55,13 +43,62 @@ def __init__(self, shape=(1,), dtype = 'float64', shm_id = None):
                 self.mmap = mmap.mmap(self._tmpFile.fileno(), self.nbytes, mmap.MAP_SHARED, mmap.PROT_READ)
                 
     def close(self):
+        """Close this buffer.
+        """
         self.mmap.close()
         if not sys.platform.startswith('win') and hasattr(self, '_tmpFile'):
             self._tmpFile.close()
     
     def to_dict(self):
-        return {'shape': self.shape, 'dtype': self.dtype, 'shm_id': self.shm_id}
+        """Return a dict that can be serialized and sent to other processes to
+        access this buffer.
+        """
+        return {'nbytes': self.nbytes, 'shm_id': self.shm_id}
+    
+    def to_numpy(self, offset, dtype, shape, strides):
+        """Return a numpy array pointing to part (or all) of this buffer.
+        """
+        return np.ndarray(buffer=self.mmap, shape=shape,
+                          strides=strides, offset=offset, dtype=dtype)        
+        
+
+
```

Comment:
I would not mind some help, but I think I am going to revert this branch back to an earlier state where it will be easier to implement swapaxes, so I don't recommend making any changes directly from here.. I'll let you know when I find a good starting place.